File: src/main/java/edu/agh/neurons/ActivationFunction.java
package edu.agh.neurons;

/**
 * Created with IntelliJ IDEA.
 * User: krzysiek
 * Date: 03.04.13
 * Time: 20:26
 * To change this template use File | Settings | File Templates.
 */
public interface ActivationFunction {

    public double activate(double in);

    public String name();

}

File: src/main/scala/edu/agh/neurons/scala/activationFunctions/ActivationFunction.scala
package edu.agh.neurons.scala.activationFunctions

/**
 * Created with IntelliJ IDEA.
 * User: krzysiek
 * Date: 24.05.13
 * Time: 19:21
 * To change this template use File | Settings | File Templates.
 */
trait ActivationFunction {
  def activate(in: Double): Double

  def name(): String
}
